### PR TITLE
PIMOB-1449: Billing form Summary: Create base button component

### PR DESCRIPTION
**Pull request by maxim-nosov-cko**
## Issue

[PIMOB-1449: Billing form Summary: Create base button component](https://checkout.atlassian.net/browse/PIMOB-1449)

## Proposed changes

Base button component was created. It should be used for all buttons and be able to fulfill all current needs.

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [x] Reviewers assigned
* [x] I have performed a self-review of my code and manual testing
* [x] Lint and unit tests pass locally with my changes
* [ ] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if applicable)

## Further comments

Samples of button design:
Screenshot 2022-10-10 at 16 44 17

Sample of enabled and disabled state:

https://user-images.githubusercontent.com/105640154/194910528-d9e97b03-f89b-403e-92cf-141197d7821c.mov

### CODE REVIEW

File: frames/src/main/java/com/checkout/frames/view/InternalButton.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+import com.checkout.frames.style.component.default.DefaultTextLabelStyle
+import com.checkout.frames.style.view.InternalButtonViewStyle
+
+@Composable
+internal fun InternalButton(
+    style: InternalButtonViewStyle,
+    state: InternalButtonState,
+    onClick: () -> Unit
+) = with(style) {
+    style.textStyle.color = Color.Unspecified
```
Comment (chintan-soni-cko) [inline frames/src/main/java/com/checkout/frames/view/InternalButton.kt:35]: Can we provide the possibility for interactionSource to interact with button events? maybe for disabling or enabling colours for it.

File: frames/src/main/java/com/checkout/frames/view/InternalButton.kt [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+        buttonElevation = ButtonElevation(defaultElevation = 8),
+        containerStyle = ContainerStyle(margin = Margin(16))
+    )
+
+    val state = stateMapper.map(style)
+
+    InternalButton(
+        styleMapper.map(style),
+        state
+    ) { state.isEnabled.value = !state.isEnabled.value }
```
Comment (chintan-soni-cko) [inline frames/src/main/java/com/checkout/frames/view/InternalButton.kt:148]: Not sure for me, how we can show the default enabled colour when the user released the button.

**Review by chintan-soni-cko — APPROVED** [review]
Few comments except LGTM. Nice job

File: frames/src/main/java/com/checkout/frames/view/InternalButton.kt (hunk 1 above)
Comment (maxim-nosov-cko) [reply frames/src/main/java/com/checkout/frames/view/InternalButton.kt:35]: `InternalButtonState` contains `isEnabled` for that purpose.

**Review by chintan-soni-cko — APPROVED** [review]

**Pull request merged**